src/Pos/Explorer/Web/Server.hs

```diff
@@ -586,15 +580,38 @@ topsortTxsOrFail f =
     maybeThrow (Internal "Dependency loop in txs set") .
     topsortTxs f
 
--- | Convert server address to client address,
--- with the possible exception (effect).
+-- | Deserialize Cardano or RSCoin address and convert it to Cardano address.
+-- Throw exception on failure.
 cAddrToAddr :: MonadThrow m => CAddress -> m Address
-cAddrToAddr cAddr = either exception pure (fromCAddress cAddr)
+cAddrToAddr cAddr@(CAddress rawAddrText) =
+    if (not $ T.null rawAddrText) && (T.last rawAddrText == '=') then do
+        -- cAddr is in RSCoin address format, converting to equivalent Cardano address
+        cardanoAddr <- convertAddr rawAddrText
+        either badRSCoinAddress pure (fromCAddress $ CAddress cardanoAddr)
+    else
+        -- cAddr is in Cardano address format
+        either badCardanoAddress pure (fromCAddress cAddr)
   where
-    exception = const $ throwM $ Internal "Invalid address!"
-
--- | Convert server transaction to client transaction,
--- with the possible exception (effect).
+    badCardanoAddress = const $ throwM $ Internal "Invalid Cardano address!"
+    badRSCoinAddress  = const $ throwM $ Internal "Invalid RSCoin address!"
+    -- Copied from cardano-sl/tools/src/keygen/Avvm.hs
+    fromAvvmPk :: (MonadThrow m, Monad m) => Text -> m RedeemPublicKey
+    fromAvvmPk addrText = do
+        let base64rify = T.replace "-" "+" . T.replace "_" "/"
```

Comment:
aha, I see you are taking base64 url form here into account.

This code https://github.com/input-output-hk/cardano-sl/blob/master/src/Pos/Wallet/Web/Server/Methods.hs#L1089 does the same thing in a more high level way - without these low level details.